src/routing/approved.rs

```diff
@@ -596,10 +553,22 @@ impl Approved {
     }
 
     // Handle `Vote` message only if signed with known key, otherwise bounce.
-    fn should_handle_vote(&self, proof_share: &ProofShare) -> bool {
-        self.section
-            .chain()
-            .has_key(&proof_share.public_key_set.public_key())
+    fn should_handle_vote(&self, vote: &Vote, proof_share: &ProofShare) -> bool {
+        // We always handle a `SectionInfo` vote for our section because it is signed by the new
+        // key produced by the last DKG and it is only about to be inserted into the chain.
+        // Any other vote must be signed by a known key.
+        match vote {
+            Vote::SectionInfo(elders_info)
```

Comment:
will this open a door that any group of adults (or even just unkowns) can send elder a Vote::SectionInfo to claim a split ?

Reply:
No, because we check whether the elders_info corresponds to the current elder candidates (the `ELDER_SIZE` oldest members) before actioning the vote (in `handle_section_info_event`).

Reply:
This is related to the below question, 
the `sender` of the vote is not checked.